sc-memory/sc-memory/src/scs/scs_parser.cpp

```diff
@@ -413,44 +456,74 @@ ElementHandle Parser::ProcessIdentifier(std::string const & name)
+  type |= scs::TypeResolver::IsConst(name) ? ScType::Const : ScType::Var;
+  return AppendElement(name, type);
 }
 
-void Parser::ProcessTriple(ElementHandle const & source, ElementHandle const & connector, ElementHandle const & target)
+void Parser::ProcessTriple(
+    ElementHandle const & sourceHandle,
+    ElementHandle const & connectorHandle,
+    ElementHandle const & targetHandle)
 {
-  ParsedElement const & connectorEl = GetParsedElement(connector);
+  ParsedElement & connector = GetParsedElementRef(connectorHandle);
+
+  auto AddConnector =
+      [this, &connector](
+          ElementHandle const & sourceHdl, ElementHandle const & connectorHdl, ElementHandle const & targetHdl)
+  {
+    ParsedElement const & source = GetParsedElement(sourceHdl);
+    ParsedElement & target = GetParsedElementRef(targetHdl);
 
-  auto AddConnector = [this, &connectorEl](ElementHandle const & src, ElementHandle const & e, ElementHandle const & trg) {
-    ParsedElement const & srcEl = GetParsedElement(src);
-    std::string const & idtf = srcEl.GetIdtf();
-    if (connectorEl.GetType() == ScType::ConstPermPosArc && scs::TypeResolver::IsKeynodeType(idtf))
+    if (source.IsElementType() && (connector.GetType() == ScType::ConstPermPosArc))
     {
-      ParsedElement & targetEl = GetParsedElementRef(trg);
-      ScType const newType = targetEl.m_type | scs::TypeResolver::GetKeynodeType(idtf);
+      std::string const & sourceIdtf = source.GetIdtf();
+      ScType const newType = target.m_type | scs::TypeResolver::GetKeynodeType(sourceIdtf);
 
-      if (targetEl.m_type.CanExtendTo(newType))
-        targetEl.m_type = newType;
+      if (target.m_type.CanExtendTo(newType))
+        target.m_type = newType;
       else
-        SC_THROW_EXCEPTION(utils::ExceptionParseError, "Can't merge types for element " + targetEl.GetIdtf());
-
-      if (!m_contourTriplesStack.empty())
-        m_parsedTriples.emplace_back(src, e, trg);
+        SC_THROW_EXCEPTION(
+            utils::ExceptionParseError,
+            "Can't extend type `" << std::string(target.m_type) << "` to type `" << std::string(newType)
+                                  << "` for element `" << target.GetIdtf() << "`.");
+
+      // TODO(NikitaZotov): Unfortunately, parser collects all sc.s-elements, and only then forms sc.s-triples based on
+      // the parsed sc.s-elements. Due to this, it is difficult to handle cases when it is necessary not to generate a
+      // sc-arc from a type of sc-elements to sc-element. Therefore, now the parser marks this arcs after that they were
+      // parsed.
+      m_elementTypeArcs.insert(connector.GetIdtf());
     }
-    else
+    else if (IsMetaElement(source) || IsMetaElement(target))
     {
-      m_parsedTriples.emplace_back(src, e, trg);
+      SC_THROW_EXCEPTION(
+        utils::ExceptionParseError,
+          "Some sc.s-triples can't be generated. Classes denoting sc-element types cannot have incoming and outgoing "
```

Comment:
ims cannot be built because of [this file](https://github.com/ostis-ai/ims.ostis.kb/blob/a7d0d8b344e2278b9d4fd30a48554b11433bddc0/ims/ostis_tech/semantic_network_represent/section_sc_code/section_subjdomain_entities/section_subjdomain_entities.scs)
and another file in this folder